platform/features/layout/test/LayoutControllerSpec.js

```diff
@@ -289,6 +317,106 @@ define(
                 expect(controller.getFrameStyle("b"))
                     .not.toEqual(oldStyle);
             });
+
+            it("allows panels to be selected", function () {
+                mockScope.$watchCollection.mostRecentCall.args[1]();
+                var childObj = mockCompositionObjects[0];
+
+                controller.select(mockEvent, childObj.getId());
+
+                expect(mockEvent.stopPropagation).toHaveBeenCalled();
+
+                expect(controller.selected(childObj)).toBe(true);
+            });
+
+            it("allows selection to be cleared", function () {
+                mockScope.$watchCollection.mostRecentCall.args[1]();
+                var childObj = mockCompositionObjects[0];
+
+                controller.select(null, childObj.getId());
+                controller.clearSelection();
+
+                expect(controller.selected(childObj)).toBeFalsy();
+            });
+
+            it("does not clear selection when moving/resizing", function () {
+                mockScope.$watchCollection.mostRecentCall.args[1]();
+                var childObj = mockCompositionObjects[0];
+                var id = childObj.getId();
+
+                controller.select(mockEvent, id);
+
+                // Do a drag
+                controller.startDrag(id, [1, 1], [0, 0]);
+                controller.continueDrag([100, 100]);
+                controller.endDrag();
+
+                // Because mouse position could cause clearSelection to be called, this should be ignored.
+                controller.clearSelection();
+
+                expect(controller.selected(childObj)).toBe(true);
+
+                // Shoud be able to clear the selection after dragging is done.
+                controller.clearSelection();
+
+                expect(controller.selected(childObj)).toBe(false);
```

Comment:
@larkin the second call to clearSelection() works as expected cause by the time it's called, the callback in settimeout (defined in endDrag) has already triggered.